src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -0,0 +1,74 @@
+package is.hail.expr.ir
+
+import is.hail.asm4s.TypeInfo
+
+import scala.language.existentials
+
+sealed case class PrimitiveOp(x: Long)
+
+sealed trait IR
+sealed case class NA(typ: PointedTypeInfo[_]) extends IR
+sealed case class I32(x: Int) extends IR
+sealed case class I64(x: Long) extends IR
+sealed case class F32(x: Float) extends IR
+sealed case class F64(x: Double) extends IR
+sealed case class True() extends IR
+sealed case class False() extends IR
+sealed case class If(cond: IR, cnsq: IR, altr: IR) extends IR
+sealed case class Let(name: String, value: IR, typ: TypeInfo[_], body: IR) extends IR
```

Comment:
I think these should have Hail types and not Java types (TypeInfo).  I feel like this is mixing levels of abstraction.